src/dragonfly/dragonfly/pump.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python3
+import sys
+import argparse
+import time
+import rclpy
+from rclpy.node import Node
+from dragonfly_messages.srv import Pump
+import rx
+import rx.operators as ops
+from rx.scheduler import NewThreadScheduler
+
+
+class BagInflateService(Node):
+    def __init__(self, arg_id, sim=False):
+        super().__init__('bag_inflate_service')
+        self.id = arg_id
+        self.sim = sim
+        if not self.sim:
+            import RPi.GPIO as GPIO
+            self.bag_gpio_pins = [16, 19, 20, 21]
+            GPIO.setmode(GPIO.BCM)
+            GPIO.setwarnings(False)
+            GPIO.setup(self.bag_gpio_pins, GPIO.OUT)
+        self.bag_full = [False] * len(self.bag_gpio_pins)
+        self.inflate = self.create_service(Pump, "/{}/pump".format(self.id), self.bag_inflate_callback)
+        # self.swap = self.create_service(BagSwap, "/{}/bagswap".format(self.id), self.bag_swap_callback)
+
+    def bag_swap_callback(self, request, response):
+        self.get_logger().info("Bags swapped")
+        self.bag_full = [False] * len(self.bag_gpio_pins)
+        return True
+
+    def bag_inflate_callback(self, request, response):
+        # @TODO add timestamp and id ect
+        self.get_logger().info("Bag inflate request received")
+        if not self.bag_full[request.pump_num]:
+            self.bag_full[request.pump_num] = True
+            if not self.sim:
+                rx.just(self.bag_gpio_pins[request.pump_num]).pipe(
+                    ops.observe_on(NewThreadScheduler())) \
+                    .subscribe(on_next=lambda pin: self.pump(pin))
+
+            response.done = True
+            return response
```

Comment:
Ok, a little nitpicky, I typically only return once at the end of the method for readability... how about setting the `response.done` in an else if the above if fails?